client/components/Dashboard.tsx

```diff
@@ -31,47 +29,243 @@ const GET_PROJECT = gql`
         project(id: $projectId) {
             id
             queries {
-                time
+                timestamp
                 depth
                 complexity
             }
         }
     }
 `;
+// we are defining state types - arrow (for sorting arrow change when clicked -ascending or descending) and time,depth and complexity props of a query
+export interface ISState {
+    style: {
+        time: boolean;
+        depth: boolean;
+        complexity: boolean;
+    };
+    arrow: {
+        time: string;
+        depth: string;
+        complexity: string;
+    };
+}
 
 function Dashboard() {
+    const [style, setStyle] = useState<ISState['style']>({
+        time: false,
+        depth: false,
+        complexity: false,
+    });
+    const [order, setOrder] = useState<boolean>(); // ascending vs descending on click
+    const [arrow, setArrow] = useState<ISState['arrow']>({
+        // adjusting arrow based on ascending vs descending
+        time: '',
+        depth: '',
+        complexity: '',
+    });
+    // toggle classes on click in the Queries component
+    const setToggle = (arg: string) => {
+        if (arg === 'time') {
+            setStyle({
+                ...style,
+                time: true,
+                depth: false,
+                complexity: false,
+            });
+        } else if (arg === 'depth') {
+            setStyle({
+                ...style,
+                time: false,
+                depth: true,
+                complexity: false,
+            });
+        } else if (arg === 'complexity') {
+            setStyle({
+                ...style,
+                time: false,
+                depth: false,
+                complexity: true,
+            });
+        }
+    };
+    // graphql calling
     const { data: dataR, loading: loadingR } = useQuery(GET_USER_DATA, {
         variables: {
-            userId: '6286978e12716d47e6884194',
+            userId: '62ba5b743f40d18829e018a1',
         },
     });
     const { data, loading } = useQuery(GET_PROJECT, {
         variables: {
-            projectId: '628e864e76cdbbec53f36010',
+            projectId: '62ba5bd43f40d18829e018c4',
```

Comment:
What is the plan for providing the `projectID` to the app when a user navigates to one of their projects? Will it be part of the url or will it be passed down as a prop?

Reply:
Passed down as a prop! Based on a user selection, the function (not currently active) called "test" (I know, poor naming, but that is not going to be the final name) will be in charge of getting the user's selection and display queries for a selected project!